packages/aws-cdk-lib/aws-ecr/lib/repository.ts

```diff
@@ -575,6 +577,14 @@ export class Repository extends RepositoryBase {
     return new Import(scope, id, {
       environmentFromArn: repositoryArn,
     });
+
+    function validateRepositoryArn() {
+      const splitArn = repositoryArn.split(':');
+
+      if (!splitArn[splitArn.length - 1].startsWith('repository/')) {
+        throw new Error(`Repository arn should be in the format 'arn:<PARTITION>:ecr:<REGION>:<ACCOUNT>:repository/<NAME>', got ${repositoryArn}.`);
```

Comment:
Commenting here that `<NAME>` cannot include `:` so there should be no valid arns that include `:` after `repository/`.